Nie można zbudować projektu - Java CI

W gałęzi `lang` podczas sprawdzania f346edba331e6e2ae0bada65c3de03fe63b218b8 wystąpił błąd budowania projektu Maven. Jest on spowodowany tym, że brakuje klas i metod, które są w gałęzi `master`. Należy zaktualizować gałąź `lang`.
[4_Build with Maven.txt](https://github.com/koder95/MiniZincDZNParser/files/4252536/4_Build.with.Maven.txt)

````log
2020-02-25T22:20:56.1722930Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/Interpreter.java:[14,51] cannot find symbol
2020-02-25T22:20:56.1724226Z   symbol:   method requireNonEmpty(java.util.Stack<java.lang.Object>)
2020-02-25T22:20:56.1725024Z   location: class pl.koder95.dznp.util.Stacks
2020-02-25T22:20:56.1727884Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/DZNParser.java:[19,59] cannot find symbol
2020-02-25T22:20:56.1728406Z   symbol:   class ParserException
2020-02-25T22:20:56.1729213Z   location: class pl.koder95.dznp.DZNParser
2020-02-25T22:20:56.1732776Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/DZNParser.java:[21,55] cannot find symbol
2020-02-25T22:20:56.1733189Z   symbol:   class ParserException
2020-02-25T22:20:56.1733945Z   location: class pl.koder95.dznp.DZNParser
2020-02-25T22:20:56.1736977Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/DZNParser.java:[23,35] cannot find symbol
2020-02-25T22:20:56.1737488Z   symbol:   class ParserException
2020-02-25T22:20:56.1738293Z   location: class pl.koder95.dznp.DZNParser
2020-02-25T22:20:56.1741252Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/DZNParser.java:[44,13] cannot find symbol
2020-02-25T22:20:56.1741610Z   symbol:   class Skipper
2020-02-25T22:20:56.1742379Z   location: class pl.koder95.dznp.DZNParser
2020-02-25T22:20:56.1745457Z [ERROR] /home/runner/work/MiniZincDZNParser/MiniZincDZNParser/src/main/java/pl/koder95/dznp/DZNParser.java:[44,35] cannot find symbol
2020-02-25T22:20:56.1745868Z   symbol:   class Skipper
2020-02-25T22:20:56.1746538Z   location: class pl.koder95.dznp.DZNParser
````

--- a/pom.xml
+++ b/pom.xml
@@ -1,34 +1,54 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
 
-  <groupId>pl.koder95</groupId>
-  <artifactId>MiniZincDZNParser</artifactId>
-  <version>0.0.1-SNAPSHOT</version>
-  <packaging>jar</packaging>
+    <groupId>pl.koder95</groupId>
+    <artifactId>dznp</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <packaging>jar</packaging>
 
-  <name>MiniZincDZNParser</name>
-  <url>http://maven.apache.org</url>
-  <distributionManagement>
-   <repository>
-     <id>github</id>
-     <name>GitHub OWNER Apache Maven Packages</name>
-     <url>https://maven.pkg.github.com/koder95/MiniZincDZNParser</url>
-   </repository>
-  </distributionManagement>
+    <name>MiniZinc DZN Parser</name>
+    <url>http://maven.apache.org</url>
+    <distributionManagement>
+        <repository>
+            <id>github</id>
+            <name>MiniZinc DZN Parser</name>
+            <url>https://maven.pkg.github.com/koder95/MiniZincDZNParser</url>
+        </repository>
+    </distributionManagement>
 
-  <properties>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.target>1.8</maven.compiler.target>
-    <maven.compiler.source>1.8</maven.compiler.source>
-  </properties>
+    <scm>
+        <connection>scm:git:git://github.com/koder95/MiniZincDZNParser.git</connection>
+        <developerConnection>scm:git:ssh://github.com:koder95/MiniZincDZNParser.git</developerConnection>
+        <url>http://github.com/koder95/MiniZincDZNParser/tree/master</url>
+    </scm>
+    <developers>
+        <developer>
+            <name>Kamil Mularski</name>
+            <email>kontakt@koder95.pl</email>
+            <organization>koder95</organization>
+            <organizationUrl>http://koder95.pl</organizationUrl>
+        </developer>
+    </developers>
+    <licenses>
+        <license>
+            <name>The Apache License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+        </license>
+    </licenses>
 
-  <dependencies>
-      <dependency>
-          <groupId>junit</groupId>
-          <artifactId>junit</artifactId>
-          <version>RELEASE</version>
-          <scope>test</scope>
-      </dependency>
-  </dependencies>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>RELEASE</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
 </project>

--- a/src/main/java/pl/koder95/dznp/ParserException.java
+++ b/src/main/java/pl/koder95/dznp/ParserException.java
@@ -1,0 +1,21 @@
+package pl.koder95.dznp;
+
+import java.io.IOException;
+
+public class ParserException extends IOException {
+    public ParserException() {
+        super("Cannot parse data!");
+    }
+
+    public ParserException(String message) {
+        super(message);
+    }
+
+    public ParserException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public ParserException(Throwable cause) {
+        super(cause);
+    }
+}

--- a/src/main/java/pl/koder95/dznp/Skipper.java
+++ b/src/main/java/pl/koder95/dznp/Skipper.java
@@ -1,0 +1,81 @@
+package pl.koder95.dznp;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class Skipper {
+
+    private final InputStream input;
+
+    public Skipper(InputStream input) {
+        this.input = input;
+    }
+
+    private int skipped = 0;
+    private int lastLastRead = -1;
+    private int lastRead = -1;
+
+    public void skip() throws IOException {
+        next();
+        while (input.available() > 0) {
+            if (lastRead == '\\') {
+                next();
+                if (isSpaceOrParagraphChar(lastRead)) lastRead = lastLastRead;
+                break;
+            }
+            if (isSpaceOrParagraphChar(lastRead)) skipSpaceCharSequence();
+            if (lastRead == '%') skipCommandSequence();
+            if (lastRead == '\n') skipNLSequence();
+            if (lastRead == '\n') break;
+        }
+    }
+
+    private void next() throws IOException {
+        this.lastLastRead = this.lastRead;
+        this.lastRead = input.read();
+    }
+
+    public int getSkipped() {
+        return skipped;
+    }
+
+    public int getLastRead() {
+        return lastRead;
+    }
+
+    private void skipSpaceCharSequence() throws IOException {
+        while (input.available() > 0) {
+            if (isSpaceOrParagraphChar(lastRead)) {
+                skipped++;
+                next();
+            }
+            else return;
+        }
+    }
+
+    private static boolean isSpaceOrParagraphChar(int lastRead) {
+        return Character.isSpaceChar(lastRead) && lastRead != '\n';
+    }
+
+    private void skipCommandSequence() throws IOException {
+        if (lastRead != '%') return;
+        if (lastRead == '%' && lastLastRead == '\\') return;
+        while (input.available() > 0) {
+            if (lastRead == '\n') return;
+            else {
+                skipped++;
+                next();
+            }
+        }
+    }
+
+    private void skipNLSequence() throws IOException {
+        if (lastRead != '\n') return;
+        while (input.available() > 0) {
+            if (lastRead == '\n') {
+                skipped++;
+                next();
+            }
+        }
+    }
+}

--- a/src/main/java/pl/koder95/dznp/util/Stacks.java
+++ b/src/main/java/pl/koder95/dznp/util/Stacks.java
@@ -22,4 +22,8 @@ public final class Stacks {
 		stack.add(obj);
 		return null;
 	}
+
+    public static <E, S extends Stack<E>> S requireNonEmpty(S stack) {
+		return stack == null || stack.empty()? null : stack;
+    }
 }

